src/Illuminate/Collections/Collection.php

```diff
@@ -514,9 +514,7 @@ public function groupBy($groupBy, $preserveKeys = false)
             foreach ($groupKeys as $groupKey) {
                 $groupKey = match (true) {
                     is_bool($groupKey) => (int) $groupKey,
-                    $groupKey instanceof \BackedEnum => $groupKey->value,
-                    $groupKey instanceof \Stringable => (string) $groupKey,
-                    default => $groupKey,
+                    default => mutate($groupKey),
                 };
```

Comment:
wouldn't this suffice?
```suggestion
               $groupKey = is_bool($groupKey) ? (int) $groupKey : mutate($groupKey);
```

Reply:
Changing the function name to `enum_value` so we need to keep `Stringable` check.

Reply:
Also reverting this changes since `illuminate/collections` doesn't require `illuminate/support`

Reply:
A little unfortunate that the scope was reduced—is there a reason, you are making this enum only?

Reply:
This allow it to cover more use case (model enum casting for example) that would failed previously if we include other types, and it makes more sense to name it `enum_value()` here.

Reply:
`transform_to_enum_value` would maybe a better name

Reply:
This method are for internal usage and not meant to be used in the user-land.

Reply:
> This allow it to cover more use case (model enum casting for example) that would failed previously if we include other types, and it makes more sense to name it `enum_value()` here.

Maybe, this can be combined, so that `mutate()` (or `toScalar()`) wraps `enum_value` (or `to_enum_value()`)